### PR TITLE
Added a SET column type. A combination of ARRAY and ENUM

### PR DESCRIPTION
**Pull request by IceShack**
I adapted the MySQL "SET" column type.
In my tests it performed as good (or bad) as that one.

There might be some missing Unit-Tests. Just let me know and I will add them.

### CODE REVIEW

File: src/Propel/Generator/Platform/MysqlPlatform.php [hunk 1]
```diff
@@ -51,6 +51,7 @@ protected function initialize()
         $this->setSchemaDomainMapping(new Domain(PropelTypes::OBJECT, 'MEDIUMBLOB'));
         $this->setSchemaDomainMapping(new Domain(PropelTypes::PHP_ARRAY, 'TEXT'));
         $this->setSchemaDomainMapping(new Domain(PropelTypes::ENUM, 'TINYINT'));
+        $this->setSchemaDomainMapping(new Domain(PropelTypes::SET, 'TINYINT'));
```
Comment (marcj) [inline src/Propel/Generator/Platform/MysqlPlatform.php]: Guess it should be int

**Comment by marcj** [comment]
Very handy new type to have multiple values in one column compared to Enum. @staabm would you like to do a review?

File: src/Propel/Generator/Builder/Om/QueryBuilder.php [hunk 2]
```diff
@@ -1051,6 +1056,37 @@ public function filterBy$colPhpName(\$$variableName = null, \$comparison = null)
...
         }";
+        } elseif ($col->isSetType()) {
+            $this->declareClasses(
+                'Propel\Common\Util\SetColumnConverter',
+                'Propel\Common\Exception\SetColumnConverterException'
+            );
+            $script .= "
+        \$valueSet = " . $this->getTableMapClassName() . "::getValueSet(" . $this->getColumnConstant($col) . ");
+        try {
+            \${$variableName} = SetColumnConverter::convertToInt(\$$variableName, \$valueSet);
```
Comment (staabm) [inline src/Propel/Generator/Builder/Om/QueryBuilder.php]: `\${$variableName}` vs. `\$$variableName` is not consistent

File: src/Propel/Generator/Model/Column.php [hunk 3]
```diff
@@ -1169,7 +1179,17 @@ public function isEnumType()
...
 
     /**
-     * Sets the list of possible values for an ENUM column.
+     * Returns whether or not this column is a SET column.
+     *
+     * @return boolean
+     */
+    public function isSetType()
+    {
+        return $this->getType() === PropelTypes::SET;
```
Comment (staabm) [inline src/Propel/Generator/Model/Column.php]: yoda notation

**Comment by staabm** [comment]
the PR is huge, but looks awesome.

**Comment by marcj** [comment]
Coolio, merged :) Thanks @IceShack!

**Pull request merged**